### PR TITLE
Quality audit - Specify the way that Volume Gizmos are being rendered

### PR DESCRIPTION
**Pull request by alex-vazquez**
---
### Purpose of this PR
- Adding user preferences to specify how the Volume Gizmos are being rendered
- Adding a system to extend Core Render Pipeliene Settings Panel
- Unify Gizmo rendering with `Volume`and `CustomPassVolume`

gizmo_visibility

---
### Testing status
- Check gizmos are correctly changed
- Check the preferences are applied, saved and restored

### CODE REVIEW

File: com.unity.render-pipelines.high-definition/Editor/RenderPipeline/CustomPass/CustomPassVolumeGizmoDrawer.cs [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+                        if (VolumesPreferences.drawSolid)
+                            Gizmos.DrawSphere(c.center, c.radius);
+                        if (scr.fadeRadius > 0)
+                            Gizmos.DrawWireSphere(c.center, c.radius + scr.fadeRadius / lossyScale.x);
+                        Gizmos.matrix = oldMatrix;
+                        break;
+                    case MeshCollider c:
+                        // Only convex mesh m_Colliders are allowed
+                        if (!c.convex)
+                            c.convex = true;
```
Comment (RSlysz) [inline com.unity.render-pipelines.high-definition/Editor/RenderPipeline/CustomPass/CustomPassVolumeGizmoDrawer.cs:68]: Same for forcing convex here

**Review by RSlysz — COMMENTED** [review]
I need more info to know if this is good or not.

File: com.unity.render-pipelines.core/Editor/Volume/VolumeGizmoDrawer.cs [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+                        if (VolumesPreferences.drawWireFrame)
+                            Gizmos.DrawWireMesh(c.sharedMesh);
+                        if (VolumesPreferences.drawSolid)
+                            // Mesh pivot should be centered or this won't work
+                            Gizmos.DrawMesh(c.sharedMesh);
+                        break;
+                    default:
+                        // Nothing for capsule (DrawCapsule isn't exposed in Gizmo), terrain, wheel and
+                        // other m_Colliders...
```
[the remark this answers is not in the document]
Comment (alex-vazquez) [reply com.unity.render-pipelines.core/Editor/Volume/VolumeGizmoDrawer.cs:75]: I think we should expose the capsule gizmo.

File: com.unity.render-pipelines.core/Editor/Volume/VolumesPreferences.cs [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+using System.Collections.Generic;
+using UnityEngine;
+using RuntimeSRPPreferences = UnityEngine.Rendering.CoreRenderPipelinePreferences;
+
+namespace UnityEditor.Rendering
+{
+    /// <summary>
+    /// Preferences for Volumes
+    /// </summary>
+    public class VolumesPreferences : ICoreRenderPipelinePreferencesProvider
```
[the remark this answers is not in the document]
Comment (alex-vazquez) [reply com.unity.render-pipelines.core/Editor/Volume/VolumesPreferences.cs:11]: For now we will leave it here, as the Gizmo menu is going to be reworked.

File: com.unity.render-pipelines.high-definition/Editor/RenderPipeline/CustomPass/CustomPassVolumeGizmoDrawer.cs [hunk 4]
```diff
@@ -0,0 +1,86 @@
+using UnityEngine;
+using UnityEngine.Rendering.HighDefinition;
+
+namespace UnityEditor.Rendering
+{
+    class CustomPassGizmoDrawer
+    {
+        [DrawGizmo(GizmoType.Active | GizmoType.Selected | GizmoType.NonSelected)]
+        static void OnDrawGizmos(CustomPassVolume scr, GizmoType gizmoType)
```
[the remark this answers is not in the document]
Comment (alex-vazquez) [reply com.unity.render-pipelines.high-definition/Editor/RenderPipeline/CustomPass/CustomPassVolumeGizmoDrawer.cs:9]: done something

File: com.unity.render-pipelines.high-definition/Editor/RenderPipeline/CustomPass/CustomPassVolumeGizmoDrawer.cs [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+                            Gizmos.DrawWireMesh(c.sharedMesh);
+                        if (VolumesPreferences.drawSolid)
+                            // Mesh pivot should be centered or this won't work
+                            Gizmos.DrawMesh(c.sharedMesh);
+
+                        // We don't display the Gizmo for fade distance mesh because the distances would be wrong
+                        break;
+                    default:
+                        // Nothing for capsule (DrawCapsule isn't exposed in Gizmo), terrain, wheel and
+                        // other m_Colliders...
```
[the remark this answers is not in the document]
Comment (alex-vazquez) [reply com.unity.render-pipelines.high-definition/Editor/RenderPipeline/CustomPass/CustomPassVolumeGizmoDrawer.cs:80]: I will wait until Gizmos API is updated.

**Review by RSlysz — APPROVED** [review]
looks better

**Review by TheoWong-pixel — APPROVED** [review]

**Comment by alex-vazquez** [comment]
QA could you check that Volume Gizmo visibility, and CustomPassVolume(HDRP) is working as is said on the PR description?

**Comment by alex-vazquez** [comment]
@reallyfancy how do you think we should include this on the documentation. I did a system to be able to add preferences to the CoreRenderPipeline page, so I think users with a custom pipeline will like to know how to do that

**Review by phi-lira — APPROVED** [review]

**Review by iM0ve — CHANGES_REQUESTED** [review]
**Testing status:** In progress

**What's tested:**


**Issues:**
**1)** On upgrade of a project it is possible the value of Gizmos is null. This will make all the volume gizmos "disappear" in the scene. 

Reproduction: need to delete registry file: Computer\HKEY_CURRENT_USER\Software\Unity Technologies\Unity Editor 5.x\General.VolumeGizmoVisibility

Unity_6Yubdxa8eF



**2)** Regression in terms of UX. It used to be very easy to temporarily disable and enable only the volume display. Now the option to do it is gone

Before:
Before

After:
After

**Review by simon-engelbrecht-soerensen — APPROVED** [review]
Working great 👍 
Data is saved and updates correctly.

**Review by iM0ve — APPROVED** [review]
I updated the review above. One issue is fixed and the other will be addressed separately
https://fogbugz.unity3d.com/f/cases/1354816/

No other concerns

**Pull request merged**